src/network/networkd-nexthop.c

```diff
@@ -549,6 +556,14 @@ static int nexthop_section_verify(NextHop *nh) {
                 /* When no Gateway= is specified, assume IPv4. */
                 nh->family = AF_INET;
 
+        if (nh->onlink < 0 && in_addr_is_set(nh->family, &nh->gw) &&
```

Comment:
Maybe add a code comment here explaining why it is enabled automatically in this one case, for extra clarity for readers